src/util.ts

```diff
@@ -896,3 +897,48 @@ export function cleanupTempDir() {
 	}
 	tmpDir = undefined;
 }
+
+/**
+ * Runs `go doc` to get documentation for given symbol
+ * @param packagePath Either the absolute path or import path of the package.
+ * @param symbol Symbol for which docs need to be found
+ * @param token Cancellation token
+ */
+export function runGodoc(packagePath: string, symbol: string, token: vscode.CancellationToken) {
+	if (!packagePath) {
+		return Promise.reject(new Error('Package Path not provided'));
+	}
+	if (!symbol) {
+		return Promise.reject(new Error('Symbol not provided'));
+	}
+
+	const goRuntimePath = getBinPath('go');
+	if (!goRuntimePath) {
+		return Promise.reject(new Error('Cannot find "go" binary. Update PATH or GOROOT appropriately'));
+	}
+
+	const cwd = packagePath;
+	const getCurrentPackagePromise = 1 !== 1 ? getCurrentPackage(cwd) : Promise.resolve(cwd);
```

Comment:
`1 !== 1`...? So `getCurrentPackage` is never really called...

I think we only need `getCurrentPackage` in the `package === ""` case from `gocode`. We can simply call it there so that this function doesn't have to call `go list` unnecessarily. Leaving this function to only handle correct package paths.

Reply:
lol... I was testing the file path case internally, and checked it in.

The `1 !== 1` was meant to be `path.isAbsolute(cwd)`

Yes, and this was for the `godef` case which returns file path always